nodejs/eks/nodes/instances.test.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+
+describe("filterEfaSubnets", () => {
+    const callReturnValues = new Map<string, pulumi.runtime.MockCallResult>();
+
+    beforeAll(() => {
+        pulumi.runtime.setMocks(
+            {
+                newResource: function (args: pulumi.runtime.MockResourceArgs): {
+                    id: string;
+                    state: any;
+                } {
+                    return {
+                        id: args.inputs.name + "_id",
+                        state: args.inputs,
+                    };
+                },
+                call: function (args: pulumi.runtime.MockCallArgs): pulumi.runtime.MockCallResult {
+                    const callReturnValue = callReturnValues.get(args.token);
+                    if (callReturnValue) {
+                        return callReturnValue;
+                    }
+
+                    return args.inputs;
+                },
+            },
+            "project",
+            "stack",
+            false, // Sets the flag `dryRun`, which indicates if pulumi is running in preview mode.
+        );
+    });
+
+    let instances: typeof import("./instances");
+    beforeEach(async function () {
+        instances = await import("./instances");
+        callReturnValues.clear();
+    });
+
+    it("should return filtered subnet IDs when the availability zone is supported", async () => {
```

Comment:
I can't remember, does our mocking library allow for us to catch and assert on errors?

Reply:
Yes, exactly! I was almost going crazy until I remembered that @t0yv0 was having issues here recently as well (https://github.com/pulumi/pulumi/issues/16665).

Had to resort to testing those errors with an e2e test :(